pype/plugins/tvpaint/publish/extract_sequence.py

```diff
@@ -0,0 +1,355 @@
+import os
+import shutil
+import tempfile
+
+import pyblish.api
+from avalon.tvpaint import lib
+
+
+class ExtractSequence(pyblish.api.Extractor):
+    label = "Extract Sequence"
+    hosts = ["tvpaint"]
+    families = ["review", "renderPass", "renderLayer"]
+
+    save_mode_to_ext = {
+        "avi": ".avi",
+        "bmp": ".bmp",
+        "cin": ".cin",
+        "deep": ".dip",
+        "dps": ".dps",
+        "dpx": ".dpx",
+        "flc": ".fli",
+        "gif": ".gif",
+        "ilbm": ".iff",
+        "jpeg": ".jpg",
+        "pcx": ".pcx",
+        "png": ".png",
+        "psd": ".psd",
+        "qt": ".qt",
+        "rtv": ".rtv",
+        "sun": ".ras",
+        "tiff": ".tiff",
+        "tga": ".tga",
+        "vpb": ".vpb"
+    }
+    sequential_save_mode = {
+        "bmp",
+        "dpx",
+        "ilbm",
+        "jpeg",
+        "png",
+        "sun",
+        "tiff",
+        "tga"
+    }
+
+    default_save_mode = "\"PNG\""
+    save_mode_for_family = {
+        "review": "\"PNG\"",
+        "renderPass": "\"PNG\"",
+        "renderLayer": "\"PNG\"",
+    }
+
+    def process(self, instance):
+        self.log.info(
+            "* Processing instance \"{}\"".format(instance.data["label"])
+        )
+
+        # Get all layers and filter out not visible
+        # TODO what to do if all are invisible?
+        # - skip without output?
+        # - still render but empty output?
+        layers = instance.data["layers"]
+        filtered_layers = [
+            layer
+            for layer in layers
+            if layer["visible"]
+        ]
+        layer_ids = [str(layer["layer_id"]) for layer in filtered_layers]
+        self.log.debug(
+            "Instance has {} layers with ids: {}".format(
+                len(layer_ids), ", ".join(layer_ids)
+            )
+        )
+        # This is plugin attribe cleanup method
+        self._prepare_save_modes()
+
+        family_lowered = instance.data["family"].lower()
+        save_mode = self.save_mode_for_family.get(
+            family_lowered, self.default_save_mode
+        )
+        save_mode_type = self._get_save_mode_type(save_mode)
+
+        is_sequence = bool(save_mode_type in self.sequential_save_mode)
+        if not is_sequence:
+            raise AssertionError((
+                "Plugin can export only sequential frame output"
+                " but save mode for family \"{}\" is not for sequence > {} <"
```

Comment:
Maybe bad naming on my side. This is checking if output extension is "sequential" -> output is not single file like `psd`, `mov, `avi`, `gif`.

BTW: I think "single frame sequence" does not work in pype plugins as should because we use `clique` to determine sequence frames which is impossible on single file.

Reply:
Righ I see what you mean. We are supporting single frame renders though. In maya at the very least, but I think that in Nuke too. TV paint should be handling them as well